auth/token.go

```diff
@@ -62,42 +67,71 @@ func NewToken(encodedToken string) (Token, error) {
 		return nil, err
 	}
 
-	return StdToken{
+	return stdToken{
 		encodedToken: encodedToken,
 		jwtToken:     decodedToken, // encapsulates jwt.token_gen from github.com/lestrrat-go/jwx/jwt
 	}, nil
 }
 
+// NewTokenFromClaims creates a Token from claims. !!! WARNING !!! No validation done when creating a Token this way. Use only in tests!
+func NewTokenFromClaims(claims map[string]interface{}) (Token, error) {
```

Comment:
Hi @f-blass It seems that this should be rather part of a test util, isn't it?

Reply:
undo is fine, closes #61

Reply:
We currently have special test util only in the mocks package. Moving the constructor there would mean changing the `stdToken` and all it's fields to public again.
I think having the constructors (see `NewToken` aswell) next to the struct is ok.

Reply:
In Java client lib, we also have the option to create a Token instance, see [here](https://github.com/SAP/cloud-security-xsuaa-integration/blob/main/java-security%2Fsrc%2Fmain%2Fjava%2Fcom%2Fsap%2Fcloud%2Fsecurity%2Ftoken%2FXsuaaToken.java#L41).
Or documented, `Token.create()`, thats ok. But whats an absolut NO-Go from clean code perspective is to expose public functions etc. for testing to productive code.

Reply:
So what would be your proposal?

Reply:
you could accept #61 and add this helper function to the test utils or where-ever it fits...

Reply:
The problem is that encodedToken and jwtToken also need to be made public then. And they can't be kept in sync if someone manually adjusts just one of the public fields

Reply:
Because of the internals of stdToken with a separate encodedToken I think it's not a good idea to make the structs field public.
To avoid this I instead needed to create a separate token struct which implements the same methods and thus also satisfies the Token interface. This one can be used to create arbitrary tokens for tests.